src/commands/auth-import.ts

```diff
@@ -0,0 +1,138 @@
+import { parse } from "csv-parse";
+import * as _ from "lodash";
+import * as Chain from "stream-chain";
+import * as clc from "cli-color";
+import * as fs from "fs-extra";
+import * as Pick from "stream-json/filters/Pick";
+import * as StreamArray from "stream-json/streamers/StreamArray";
+
+import { Command } from "../command";
+import { FirebaseError } from "../error";
+import { logger } from "../logger";
+import { needProjectId } from "../projectUtils";
+import { requirePermissions } from "../requirePermissions";
+import * as accountImporter from "../accountImporter";
+import * as utils from "../utils";
+
+const MAX_BATCH_SIZE = 1000;
+const validateOptions = accountImporter.validateOptions;
+const validateUserJson = accountImporter.validateUserJson;
+const transArrayToUser = accountImporter.transArrayToUser;
+const serialImportUsers = accountImporter.serialImportUsers;
+
+module.exports = new Command("auth:import [dataFile]")
+  .description("import users into your Firebase project from a data file(.csv or .json)")
+  .option(
+    "--hash-algo <hashAlgo>",
+    "specify the hash algorithm used in password for these accounts"
+  )
+  .option("--hash-key <hashKey>", "specify the key used in hash algorithm")
+  .option(
+    "--salt-separator <saltSeparator>",
+    "specify the salt separator which will be appended to salt when verifying password. only used by SCRYPT now."
+  )
+  .option("--rounds <rounds>", "specify how many rounds for hash calculation.")
+  .option(
+    "--mem-cost <memCost>",
+    "specify the memory cost for firebase scrypt, or cpu/memory cost for standard scrypt"
+  )
+  .option("--parallelization <parallelization>", "specify the parallelization for standard scrypt.")
+  .option("--block-size <blockSize>", "specify the block size (normally is 8) for standard scrypt.")
+  .option("--dk-len <dkLen>", "specify derived key length for standard scrypt.")
+  .option(
+    "--hash-input-order <hashInputOrder>",
+    "specify the order of password and salt. Possible values are SALT_FIRST and PASSWORD_FIRST. " +
+      "MD5, SHA1, SHA256, SHA512, HMAC_MD5, HMAC_SHA1, HMAC_SHA256, HMAC_SHA512 support this flag."
+  )
+  .before(requirePermissions, ["firebaseauth.users.create", "firebaseauth.users.update"])
+  .action(async (dataFile, options) => {
```

Comment:
Can we mkae this `dataFile: string, options: Options` so some of these linter warnings go away?

Reply:
Done.